External/YAeHMOP/EHTTools.cpp

```diff
@@ -25,6 +27,42 @@ const std::string _EHTChargeMatrix = "_EHTChargeMatrix";
 std::mutex yaehmop_mutex;
 #endif
 
+namespace {
+std::string randomstring(unsigned int len = 8) {
+  const std::string alphanum{
+      "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"};
+  const auto strSize = alphanum.size();
+  std::string res;
+  res.reserve(len);
+
+  while (res.size() != len) {
+    const auto pos = rand() % strSize;
+    res.push_back(alphanum[pos]);
+  }
+  return res;
+}
+// RAII tempfile class
+struct Tempfile {
+  std::filesystem::path fname;
+  FILE *fptr;
+
+  Tempfile() {
+    fname = std::filesystem::temp_directory_path() / randomstring();
+    fptr = std::fopen(fname.c_str(), "w+");
```

Comment:
Windows paths are wchar_t so can't be used like this in std::fopen

You might need to use _wfopen

```
fptr = _wfopen(fname..c_str(),  L"w+");
```

 or try and convert to char:

```
fptr = std::fopen(fname.string().c_str(), "w+");
```

Reply:
https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/fopen-wfopen?view=msvc-170

Reply:
done